### PR TITLE
Feature/co 487 update api to use logger

### PR DESCRIPTION
**Pull request by gmtcrary**
PR for CO-487.
This PR adds the IOExceptionMapper and GenericExceptionMapper classes and registers them to the jersey environment.

Uncaught exceptions that bubble up past the resource endpoint level (of the stack) get caught by the registered exception mappers.

*The Exception handler demo has been removed as of [5e3182e](https://github.com/osu-mist/web-api-skeleton/pull/43/commits/5e3182edf382f4d384353ae1f4d9d5ecf1b92296)

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy [hunk 1]
```diff
@@ -26,6 +26,12 @@ class InfoResource extends Resource {
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     public Response getInfo(@Auth AuthenticatedUser authenticatedUser) {
+        try {
+            throw new IllegalFormatPrecisionException(3)
+        } catch(FileNotFoundException e) {
+        }
+
```
Comment (jaredkosanovic) [inline src/main/groovy/edu/oregonstate/mist/api/InfoResource.groovy]: A few too many blank lines here

File: src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Created by georgecrary on 8/5/16.
+ */
+@Provider
+class GenericExceptionMapper implements ExceptionMapper<Exception> {
+    Logger logger = LoggerFactory.getLogger(GenericExceptionMapper.class)
+
+    @Override
+    public Response toResponse(Exception e) {
+        logger.error("Generic exception occured and caught by GEMapper. Please map $e" ,e)
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy]: Don't abbreviate GenericExceptionMapper in this message.
What does please map mean?
s/\$e/{}/ or s/ ,e//

File: src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.ws.rs.core.Response
+import javax.ws.rs.ext.ExceptionMapper
+import javax.ws.rs.ext.Provider
+
+/**
+ * Created by georgecrary on 8/5/16.
+ */
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy]: Your name and the date are already attached to the commit. It isn't necessary to pepper source files with these kind of comments.

File: src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    @Override
+    public Response toResponse(Exception e) {
+        logger.error("Generic exception occured and caught by GEMapper. Please map $e" ,e)
+
+        Response.ResponseBuilder builder = Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+        builder.entity(new Error(
+                status: Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                developerMessage: "Generic exception caught and mapped.",
+                userMessage: "Generic Exception. Please contact a dev.",
+                code: Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy]: I think this 'code' is supposed to correspond with the 4th column on https://developer.oregonstate.edu/documentation/error-reference

File: src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy [hunk 5]
```diff
@@ -25,17 +27,32 @@ class SkeletonApplication extends Application<Configuration> {
...
      * Parses command-line arguments and runs the application.
      *
      * @param configuration
      * @param environment
      */
     @Override
     public void run(Configuration configuration, Environment environment) {
-        Resource.loadProperties()
-
+        Resource.loadProperties(environment)
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy]: Does this actually work? Resource.loadProperties doesn't take any arguments.

File: src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy [hunk 6]
```diff
@@ -25,17 +27,32 @@ class SkeletonApplication extends Application<Configuration> {
     public void initialize(Bootstrap<Configuration> bootstrap) {}
 
     /**
+     * Registers lifecycle managers and Jersey exception mappers
+     *
+     * @param environment
+     * @param buildInfoManager
+     */
+    protected void registerAppManagerLogic(Environment environment,
```
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/webapiskeleton/SkeletonApplication.groovy:35]: registerAppManagerLogic doesn't seem like a great name (it does more than register the lifecycle manager, and 'logic' is superfluous), but i'm not sure i can think of a better one. skeletonSetup? Would it make sense to register the InfoResource endpoint here as well?

File: src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy (hunk 2 above)
Comment (magical) [inline src/main/groovy/edu/oregonstate/mist/api/jsonapi/GenericExceptionMapper.groovy]: > s/\$e/{}/ or s/ ,e//

Never mind. I was looking at the documentation for the wrong error method.

**Pull request merged**